### ISSUE TITLE
Faucet has hardcoded 200 tokens to distribution, fails if total below that.

### ISSUE THREAD

**Issue by ethanfrey** [CosmWasm/cosmjs#369]
Faucet is working now, but it doesn't work with such low values:
I put 100SHELL in the distribution account, as we give out 1 SHELL per user.

```
Aug 11 10:55:52 faucet docker[31029]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral12x4sdtlt2zg9rexp405saahs3dellm>
Aug 11 10:55:52 faucet docker[31029]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1r8k9dwsryvuq6k05tgtwxzdxkwq7up>
Aug 11 10:55:52 faucet docker[31029]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1vr7lum34zwfz4w9meyyc4s9ce4k70h>
Aug 11 10:55:52 faucet docker[31029]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1fp2ntmn3f5vewyd0g3vlpqtmqy39ug>
Aug 11 10:55:52 faucet docker[31029]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1ucpr9rwudeaq0q2ux8a5gywx3ndrxu>
Aug 11 10:55:52 faucet docker[31029]: Done refilling accounts.
Aug 11 10:55:52 faucet docker[31029]: Holder:
Aug 11 10:55:52 faucet docker[31029]:   coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc: [100 SHELL]
Aug 11 10:55:52 faucet docker[31029]: Distributors:
Aug 11 10:55:52 faucet docker[31029]:   coral12x4sdtlt2zg9rexp405saahs3dellmwa76s4ce: []
Aug 11 10:55:52 faucet docker[31029]:   coral1r8k9dwsryvuq6k05tgtwxzdxkwq7up3ngqpnhy: []
Aug 11 10:55:52 faucet docker[31029]:   coral1vr7lum34zwfz4w9meyyc4s9ce4k70hqcldz5ee: []
Aug 11 10:55:52 faucet docker[31029]:   coral1fp2ntmn3f5vewyd0g3vlpqtmqy39ug9rph7gxq: []
Aug 11 10:55:52 faucet docker[31029]:   coral1ucpr9rwudeaq0q2ux8a5gywx3ndrxusa7lvm62: []
```

I can work around this by sending 1000SHELL to the account, but maybe we can make this more dynamic (checking the token holders balance). Or maybe we can just use more tokens in the testnets.

**Comment by ethanfrey** [comment]
Actually, I gave 1000SHELL to the token holder and still get this error:

```
Aug 11 10:59:31 faucet docker[31180]: Connected to network: cosmwasm-coral
Aug 11 10:59:31 faucet docker[31180]: Tokens on network: SHELL, REEF
Aug 11 10:59:31 faucet docker[31180]: Holder:
Aug 11 10:59:31 faucet docker[31180]:   coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc: [1000 SHELL]
Aug 11 10:59:31 faucet docker[31180]: Distributors:
Aug 11 10:59:31 faucet docker[31180]:   coral12x4sdtlt2zg9rexp405saahs3dellmwa76s4ce: []
Aug 11 10:59:31 faucet docker[31180]:   coral1r8k9dwsryvuq6k05tgtwxzdxkwq7up3ngqpnhy: []
Aug 11 10:59:31 faucet docker[31180]:   coral1vr7lum34zwfz4w9meyyc4s9ce4k70hqcldz5ee: []
Aug 11 10:59:31 faucet docker[31180]:   coral1fp2ntmn3f5vewyd0g3vlpqtmqy39ug9rph7gxq: []
Aug 11 10:59:31 faucet docker[31180]:   coral1ucpr9rwudeaq0q2ux8a5gywx3ndrxusa7lvm62: []
Aug 11 10:59:31 faucet docker[31180]: Available tokens: [ 'SHELL' ]
Aug 11 10:59:31 faucet docker[31180]: Refilling SHELL of:
Aug 11 10:59:31 faucet docker[31180]:   coral12x4sdtlt2zg9rexp405saahs3dellmwa76s4ce: []
Aug 11 10:59:31 faucet docker[31180]:   coral1r8k9dwsryvuq6k05tgtwxzdxkwq7up3ngqpnhy: []
Aug 11 10:59:31 faucet docker[31180]:   coral1vr7lum34zwfz4w9meyyc4s9ce4k70hqcldz5ee: []
Aug 11 10:59:31 faucet docker[31180]:   coral1fp2ntmn3f5vewyd0g3vlpqtmqy39ug9rph7gxq: []
Aug 11 10:59:31 faucet docker[31180]:   coral1ucpr9rwudeaq0q2ux8a5gywx3ndrxusa7lvm62: []
Aug 11 10:59:31 faucet docker[31180]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral12x4sdtlt2zg9rexp405saahs3dellm>
Aug 11 10:59:31 faucet docker[31180]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1r8k9dwsryvuq6k05tgtwxzdxkwq7up>
Aug 11 10:59:31 faucet docker[31180]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1vr7lum34zwfz4w9meyyc4s9ce4k70h>
Aug 11 10:59:31 faucet docker[31180]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1fp2ntmn3f5vewyd0g3vlpqtmqy39ug>
Aug 11 10:59:31 faucet docker[31180]: Sending 200 SHELL from coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc to coral1ucpr9rwudeaq0q2ux8a5gywx3ndrxu>
Aug 11 10:59:31 faucet docker[31180]: Done refilling accounts.
Aug 11 10:59:31 faucet docker[31180]: Holder:
Aug 11 10:59:31 faucet docker[31180]:   coral1nkwhvtmdu5dm0yd4a8xvml3ktpdg9lhsqavygc: [1000 SHELL]
Aug 11 10:59:31 faucet docker[31180]: Distributors:
Aug 11 10:59:31 faucet docker[31180]:   coral12x4sdtlt2zg9rexp405saahs3dellmwa76s4ce: []
```

This is with cosmjs 0.22.1. Configuration:

```
FAUCET_CREDIT_AMOUNT_SHELL=1
FAUCET_ADDRESS_PREFIX=coral
FAUCET_TOKENS="SHELL=10^6ushell,REEF=10^6ureef"
```

**Comment by webmaster128** [comment]
The 200 SHELL shell is `FAUCET_REFILL_FACTOR` (defaults to 20) multiplied by `FAUCET_CREDIT_AMOUNT_SHELL` (defaults to 10). Looks like `FAUCET_CREDIT_AMOUNT_SHELL=1` was not set or there is a bug.

What I find more annoying is the lack of error messages in case of send problems. It might get lost in an async context.

The core bug is most likely the usage of ucosm fees.

**Comment by ethanfrey** [comment]
You are right, the number issue was poorly passing it. I changed the variable name from `FAUCET_CREDIT_AMOUNT_COSM` to `FAUCET_CREDIT_AMOUNT_SHELL` one place, but then forgot in the other, so it was not getting any value. I fixed that in the config and it tries to send 20.

The sends still fail though

**Comment by webmaster128** [comment]
The fix #370 was now merged into 0.22 and released as 0.22.2
